lib/jekyll/collection.rb

```diff
@@ -212,7 +212,9 @@ def container
     def read_document(full_path)
       doc = Document.new(full_path, :site => site, :collection => self)
       doc.read
-      docs << doc unless doc.data["published"] == false
+      unless doc.data["published"] == false && !site.unpublished
```

Comment:
This `unless` makes my head hurt.

"**Unless** **not** the site is **un**published"

Reply:
```ruby
if site.unpublished || doc.data["published"] != false
  docs << doc
end
```

❓

Reply:
Reversing the conditional leads us to another cleaner alternative:
```ruby
if site.unpublished || doc.published?
  docs << doc
end
```

Reply:
Agreed, just pushed up a commit that amends this.